kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/codec/KafkaMessageDecoder.java

```diff
@@ -31,6 +34,10 @@ public void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) {
             try {
                 int sof = in.readerIndex();
                 int frameSize = in.readInt();
+                if (frameSize > socketFrameMaxSize) {
+                    throw new RuntimeException(
```

Comment:
I'd suggest throwing a FrameOversizedException and giving the responsibility to report the hint to a handler (ChannelHandler#exceptionCaught) installed in the request pipeline.